extension/src/popup/components/manageAssets/SearchAsset/hooks/useAssetLookup.ts

```diff
@@ -0,0 +1,373 @@
+import { useReducer } from "react";
```

Comment:
this pulls out all the logic for looking up assets based on type into a reusable hook. This hook can be dropped into a shared lib for use in both extension and mobile